### PR TITLE
DOCSP-43159: QB returns objects

### PR DESCRIPTION
**Pull request by rustagir**
adds documentation to note that QB returns results as objects instead of arrays in v5

https://jira.mongodb.org/browse/DOCSP-43159

[Staging](https://preview-mongodbrustagir.gatsbyjs.io/laravel/DOCSP-43159/query-builder/#overview)
[upgrade guide](https://preview-mongodbrustagir.gatsbyjs.io/laravel/DOCSP-43159/upgrade/#version-5.x-breaking-changes)

### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file

### CODE REVIEW

**Review by jordan-smith721 — APPROVED** [review]
LGTM

**Comment by GromNaN** [comment]
This needs to be documented in the upgrade guide. This is a major breaking change.

**Review by norareidy — APPROVED** [review]

File: docs/upgrade.txt [hunk 1]
```diff
@@ -71,6 +71,21 @@ Version 5.x Breaking Changes
...
+  The following code shows how to retrieve a query result and access a
+  property from the result object:
+
+  .. code-block:: php
+     :emphasize-lines: 4
+
+     DB::table('accounts')
+         ->where('name', 'Anita Charles')
+         ->first()
+         ->balance
```
Comment (GromNaN) [inline docs/upgrade.txt]: You could highlight the change with a diff (not sure if diffs are correctly colorized in the docs):

```suggestion
     $document = DB::table('accounts')
         ->where('name', 'Anita Charles')
         ->first();

-    $document['balance'];
+    $document->balance;
```

Preview:
```diff
     $document = DB::table('accounts')
         ->where('name', 'Anita Charles')
         ->first();

-    $document['balance'];
+    $document->balance;
```

File: docs/query-builder.txt [hunk 2]
```diff
@@ -38,15 +38,20 @@ testability.
 
 The {+odm-short+} provides the ``DB`` method ``table()`` to access a collection.
 Chain methods to specify commands and any constraints. Then, chain
-the ``get()`` method at the end to run the methods and retrieve the results.
+the ``get()`` method at the end to run the methods and retrieve the
+results. To retrieve only the first matching result, chain the
+``first()`` method instead of the ``get()`` method. Starting in
+{+odm-long+} v5.0, the query builder returns results as objects.
```
Comment (GromNaN) [inline docs/query-builder.txt]:
```suggestion
{+odm-long+} v5.0, the query builder returns results as ``stdClass`` objects.
```

File: docs/upgrade.txt (hunk 1 above)
Comment (rustagir) [reply docs/upgrade.txt]: The only way we can demonstrate this is by using code comments. I can make that change though!

**Review by GromNaN — APPROVED** [review]
LGTM

**Pull request merged**